### DISCUSSION TITLE
What's the best Upgrade method these days?

### DISCUSSION THREAD

**Discussion by Bugman1400** [aqualinkd/AqualinkD#370] [General]
One of the unfortunate aspects of this software is that its so reliable that I forget what I had to do the last time. ;)
The Upgrade section indicates that I can just do a 'git pull', but I get a 'fatal: not a git repository (or any of the parent directories): .git' error message. I would rather not have to re-configure a brand new installation. I don't have the Docker, just the AqualinkD service. I have version 2.3.7 currently and would like to Upgrade to 2.5.1 to test out the scheduling after a POWER ON or FREEZE PROTECT event.

Any suggestions?

**Comment by sfeakes** [comment]
As long as you run the aqualinkd install script that's within the repo, it will NOT overwrite any of your configuration files. So you shouldn't have to worry about that.  But getting that install script in your system is the problem you are having at the moment.
BTW.  Making a copy / backup of the below files is all you need to do, to keep your current config safe.
```
/etc/aqualinkd.conf
/var/www/aqualinkd/config.js
```

But to the real problem.  It would depend on how you initially installed AqualinkD as to how you need to update it. Also 2.5.1 is in development and not a final release yet, so you'd need to force aqualinkd to pull a dev version and not release version (it will pull release by default).

The below is probably the easiest way to do that.
```
cd ~
curl -s https://raw.githubusercontent.com/sfeakes/AqualinkD/master/extras/aqua.sh -o aqua.sh 
chmod 755 aqua.sh
./aqua.sh force_latest
```

When 2.5.1 becomes a release version the command(s) are almost the same, just the last one changes.
```
cd ~
curl -s https://raw.githubusercontent.com/sfeakes/AqualinkD/master/extras/aqua.sh -o aqua.sh 
chmod 755 aqua.sh
./aqua.sh release
```

The `git pull` probably didn't work for one of two reasons. You didn't use git to install AqualinkD originally, or if you did you were not in the same directory where you pulled it previously.

**Reply by sfeakes** [reply]
Forgot to say.   Very glad AqualinkD has been stable / reliable for you, that's always good to hear.

You will need to add the below to /etc/aqualinkd.conf for the new scheduling event options
```
# Check if button_01 (usually Pump) is scheduled to run after an event that may have turned it off, and set it to run.
# Only for RS panels, Will not work for PDA panles.
# Example below is if pump is off due to power reset, freezeprotect or swg boots is turned off between 6am and 11pm then turn the pump on.
# You can leave scheduler_check_pumpon_hour & scheduler_check_pumpoff_hour commented out and AqualinkD will try to find the hours from the actual schedule
# that's been set in scheduler.  This will only work if you have the same schedule for every day of the week. 
#scheduler_check_pumpon_hour = 6
#scheduler_check_pumpoff_hour = 23
scheduler_check_poweron = yes
scheduler_check_freezeprotectoff = yes
scheduler_check_boostoff = yes
```

**Reply by Bugman1400** [reply]
> Forgot to say. Very glad AqualinkD has been stable / reliable for you, that's always good to hear.
> 
> You will need to add the below to /etc/aqualinkd.conf for the new scheduling event options
> 
> ```
> # Check if button_01 (usually Pump) is scheduled to run after an event that may have turned it off, and set it to run.
> # Only for RS panels, Will not work for PDA panles.
> # Example below is if pump is off due to power reset, freezeprotect or swg boots is turned off between 6am and 11pm then turn the pump on.
> # You can leave scheduler_check_pumpon_hour & scheduler_check_pumpoff_hour commented out and AqualinkD will try to find the hours from the actual schedule
> # that's been set in scheduler.  This will only work if you have the same schedule for every day of the week. 
> #scheduler_check_pumpon_hour = 6
> #scheduler_check_pumpoff_hour = 23
> scheduler_check_poweron = yes
> scheduler_check_freezeprotectoff = yes
> scheduler_check_boostoff = yes
> ```

I don't use the AqualinkD scheduler (even though its a great feature). I use home automation (HomeSeer) to turn the main pump and pool light on/off via MQTT. I have a virtual latch set in HomeSeer that should turn the main pump back on after a FREEZE PROTECT event ends, but it doesn't. Is AqualinkD sending/receiving the correct MQTT messages after a FREEZE PROTECT event? Perhaps I need to look for the FREEZE PROTECT message as well and trigger off of that. Not sure what to do for a POWER ON event since there is no MQTT message for that.

**Reply by sfeakes** [reply]
Absolutely it will work with HomeSeer (or HA) or anything else you use to schedule.  But in those cases where you don't use the internal scheduler, you will absolutely need to use the `scheduler_check_pumpon_hour = ` and `scheduler_check_pumpoff_hour = ` options.

BTW, I just posted 2.5.1 as a release (no longer dev).
